autoload/hackcomplete.vim

```diff
@@ -41,7 +41,7 @@ function! hackcomplete#Complete(findstart, base)
 
   " Insert the base and magic token into the current line.
   let curline = lines[lnum - 1]
-  let lines[lnum - 1] = curline[:cnum - 1] . a:base . s:autotok . curline[cnum :]
+  let lines[lnum - 1] = curline[:cnum - 2] . a:base . s:autotok . curline[cnum - 1] . curline[cnum :]
```

Comment:
I also am not a vimscript expert but it looks like this should be equivalent to:

`let lines[lnum - 1] = curline[:cnum - 2] . a:base . s:autotok . curline[cnum - 1 :]`